tests/test_transformation.py

```diff
@@ -146,3 +149,63 @@ def test_does_pdf_contain_cmyk(client, data, result):
 )
 def test_does_pdf_contain_rgb(client, data, result):
     assert does_pdf_contain_rgb(BytesIO(data)) == result
+
+
+def detect_color_space(page):
+    if does_pdf_contain_cmyk(page):  # CMYK images have 4 color channels
+        return "CMYK"
+    return "RGB"
+
+
+def compare_images(page1, page2):
```

Comment:
Would be tempted to call this something like `are_images_different` or something along those lines.